spsa.py

```
@@ -1,454 +1,422 @@
+    lines.append("} // namespace engine::eval")
+    lines.append("#endif")
+    
+    # Also write to the default location
+    with open("Weights.h", "w", encoding="utf-8") as f:
+        f.writelines(l + "\n" for l in lines)
+# --- SPSA Core ---
+def spsa_core(params, args):
+    for k in range(args.iters):
+        logger.info(f"=== iter: {k} ===")
+
+        deltas = {n: (1 if random.random() < 0.5 else -1) for n in params}
```

Ruff (0.15.21)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)